docs/index.md

```diff
@@ -210,8 +210,17 @@ resource "azurerm_databricks_workspace" "this" {
   sku                           = "premium"
 }
 
+data "azurerm_databricks_workspace" "this" {
```

Comment:
the examples do work most of the times like they were before. This PR seems to suggest a workaround with data resource, but does not explain why.

Please do not modify these examples and instead add a new paragraph with detailed explanation, issue reference (https://github.com/hashicorp/terraform/issues/2430), and code example. Make sure to modify azure workspace guide as well.